### PR TITLE
Email confirmation feature

### PR DESCRIPTION
**Pull request by d0peCode**
Added email confirmation feature:
 - credentails needed to send email are in .env.examle and used in `config/index.js`
- `README.md` udpated to additional tips about setting transporter
- Added `uuid` package to create unique key for account activation
- Created `transporter.js` in services
- Added `active: Boolean` and `activationKey: String` in user model
- Added `userSchema.post('save', async function` with function which is sending email after save in user.model file (below .pre hook)
- Added `if (!user.active) throw new APIError(`User not activated`, httpStatus.UNAUTHORIZED)` in user.model file within `findAndGenerateToken` function, in order to not let inactive user login.
- Added `/confirm` route which is invoked after clicking link from mail.

Please do code review. Especially things I don't know are:

- in `userSchema.method({` there is this array `const fields = ['id', 'name', 'email', 'createdAt', 'role']` shoudn't it contain two new fields (`active` and `activationKey`)?
- in `.pre` hook there is this condition:

    ```
    if (!this.isModified('password')) {
        return next()
    }
which I don't have in `.post` hook. Should it be there?

### CODE REVIEW

**Comment by kasvith** [comment]
@d0peCode can you please check why travis build fails?

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,6 @@
 node_modules/
 package-lock.json
 .env
+.idea
+.env
+.env-example
```
Comment (kasvith) [inline .gitignore]: add a new line

File: README.md [hunk 2]
```diff
@@ -5,6 +5,28 @@
...
 - Check `http://localhost:3000/api/status` to see it works
+
+## Note when setting transporter in `.env`
+
+- TRANSPORTER_SERVICE can be for example `gmail` but also SMTP of your server
+- TRANSPORTER_EMAIL is email from which mails will be sent
+- TRANSPORTER_PASSWORD is password to above email 
+  - if gmail you need to generate special app-password, see for further support: https://support.google.com/mail/answer/185833?hl=en)
+
+## Changes from original project
```
Comment (kasvith) [inline README.md]: From this line onwards is not needed for readme

**Comment by kasvith** [comment]
@d0peCode can you pull from master? made some changes to fix travis

**Comment by d0peCode** [comment]
> Merge remote-tracking branch 'upstream/master' into emailConfirmation

Travis is still failing, I don't know why.

**Comment by kasvith** [comment]
Now it works 💯

**Comment by kasvith** [comment]
Thanks, @d0peCode you did a great job

**Pull request merged**